NEWS.md

```diff
@@ -2,7 +2,16 @@ MXNet Change Log
+An incorrect [usage](https://github.com/apache/incubator-mxnet/issues/11091) of `save_params` was advertised in the gluon book which led MXNet users to depend on the incorrect usage and developing a hack around it. A change was made to the internal structure of the `.params` file saved by `save_params` to resolve a bug. This led to user scripts with the above mentioned hack to break. To fix this, `save_params` and `load_params` APIs have been reverted to previous format and marked as deprecated. New APIs: `save_parameters` and `load_parameters` have been added for the new format. All scripts to save and load parameters for a Gluon model should now use the new API for `save_parameters` and `load_parameters`. If your model is hybridizable and you want to export a serialized structure of the model as well as parameters you need to use the `export` API and the newly added `imports` API instead of `save_params` and `load_params` API. 
+
+### User Code Changes
+- If you have been using the `save_params` and `load_params` API, below are the recommendations on how to update your code:
+1. If you save parameters to load it back into a `SymbolBlock`, it is strongly recommended to use `export` and `imports` API instead. For more information, please see the [Saving and Loading Gluon Models Tutorial](https://mxnet.incubator.apache.org/tutorials/gluon/save_load_params.html).
+2. If you created gluon layers without a `name_scope` using MXNet 1.2.0, you must replace `save_params` with `save_parameters`. Otherwise, your models saved in 1.2.1 will fail to load back, although this worked in 1.2.0.
+3. For the other use cases, such as models created within a `name_scope` (inside a `with name_scope()` block) or models being loaded back into gluon and not `SymbolBlock`, we strongly recommend replacing `save_params` and `load_params` with `save_parameters` and `load_parameters`. Having said that, your code won't break in 1.2.1 but will give you a deprecated warning message for `save_params` and `load_params`.
+
+### Exception to Semantic Versioning
```

Comment:
I don't think we need to explain we are making exception to SemVer. Why is it important for the user to know as a part especially as a part of Release notes

Reply:
This was suggested by @ThomasDelteil and the reasoning was that some users may depend on version numbers to decide whether it is a breaking change or an addition or a bug fix based on versioning.

Reply:
@nswamy violation of our own rule without even explanation seems like a good way of losing trust.

Reply:
I understand @nswamy 's point though. There may be users who don't understand what  semantic versioning and for them this piece of text is just confusing also given that our earlier releases never had it. Which group of users is bigger is debatable and I don't have data to prove either way.

Reply:
Then why don't we make the text more self-contained by including a link to the definition of semantic versioning? This is about our own stance, and not at all about group sizes.

Reply:
There is already enough detail on what the breaking changes and how to use it going forward. I don't see a reason why we should overwhelm the user with these details.

Reply:
@szha there is no need to talk about semver in the release notes.

Reply:
@szha this is not about hiding anything..I am suggesting we keep the release notes just enough.

Reply:
@nswamy let's try to work towards a compromise here. I think there are valid concerns that originated from @szha. I think factually it is true that we are doing an exception to semantic versioning, and I think it is important for our users to see that we know it and take this seriously and are transparent about it, to increase the trust in the future release.

@nswamy Do you feel strongly that this should be removed because I think some others in the discussion really feel strongly that it should stay. If it's just that you think it is adding unnecessary details, I would suggest to leave it as it is, and would suggest removing your veto. For transparency and generally documentation cases, I think it is better to have too much than too little.

Reply:
@ThomasDelteil Release notes is not a place to add these unnecessary information. If people feel strongly raise it else where, why are we burdening the user to read abut SemVer, why is it important for the user to know why you made an exception. 
I would like this information to be dropped from the Release notes for our Users sake, there will no end to what goes in release notes.

Reply:
@ThomasDelteil @szha adding a **Breaking Change** section is more appropriate in the Release Notes(which I think is very well explained already)

Reply:
It might be rephrased, but I agree that we need something with justification to express that this was a one-time-off (other frameworks make a lot of breaking changes with minor releases). Having a stable API gives us a lot trust and is often a major point for users. We had customer using 0.11 although 1.1 was released just because they were afraid of upgrading. Same for 1.0 users although 1.2 is released - we should make it our highest priority that we gain trust in our versioning and explain it in quite detail if we diverge from it. Ideally, we'd be like Apple and basically have everybody upgrade as soon as an update gets released. Otherwise, we'll always have to keep supporting old versions although we officially say that we don't.